code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -197,6 +209,32 @@ void setNetwork(Map<String, Object> inspect, Instance instance) {
         }
     }
 
+    boolean isIpInNetwork(String ipAddress, Network network) {
+        for (Subnet subnet : objectManager.children(network, Subnet.class)) {
+            if (subnet.getRemoved() == null) {
+                String startIp = NetUtils.getDefaultStartAddress(subnet.getNetworkAddress(), subnet.getCidrSize());
+                long start = NetUtils.ip2Long(startIp);
+                String endIp = NetUtils.getDefaultEndAddress(subnet.getNetworkAddress(), subnet.getCidrSize());
+                long end = NetUtils.ip2Long(endIp);
+                long ip = NetUtils.ip2Long(ipAddress);
+                if (start < ip && ip <= end)
```

Comment:
Sorry, I don't know if this was clear, but does the last condition need to be `<`?  Does defaultEndAddress return `x.x.x.255`?  That IP should be allowed either.